src/nmodl/noccout.cpp

```diff
@@ -64,6 +68,108 @@ static void ext_vdef() {
+            P("    auto* const _p = _pnt->_prop;\n");
+            P("    _nrn_mechanism_cache_instance _ml_real{_p};\n");
+            P("    auto* const _ml = &_ml_real;\n");
+            P("    size_t const _iml{};\n");
+            P("    auto* _ppvar = _nrn_mechanism_access_dparam(_p);\n");
+            P("    auto* _thread = _extcall_thread.data();\n");
+            P("    auto* _nt = static_cast<NrnThread*>(_pnt->_vnt);\n");
+        } else {
+            P("    Prop* _local_prop = _extcall_prop;\n");
+            P("    _nrn_mechanism_cache_instance _ml_real{_local_prop};\n");
+            P("    auto* const _ml = &_ml_real;\n");
+            P("    size_t const _iml{};\n");
+            P("    auto* _ppvar = _local_prop ? _nrn_mechanism_access_dparam(_local_prop) : nullptr;\n");
+            P("    auto* _thread = _extcall_thread.data();\n");
+            P("    auto* _nt = nrn_threads;\n");
+        }
+    };
+
+    // Emit HPC wrapper for init_rng function
+    // TODO: this assumes we are using RANDOM in POINT_PROCESS.
+    //
+    P("static double _hoc_init_rng(void* _vptr) {\n");
+    common_code_for_hoc_function();
+    // TODO: once we eliminate the first argument as variable name, there should be 3 arguments
+    P("    if (!ifarg(4)) {\n");
+    P("        hoc_execerror(\"init_rng() requires 4 arguments!\", NULL);\n");
+    P("    }\n");
+    P("    const char *_name = gargstr(1);\n");
+    P("    uint32_t _id1 = uint32_t(*getarg(2));\n");
+    P("    uint32_t _id2 = uint32_t(*getarg(3));\n");
+    P("    uint32_t _id3 = uint32_t(*getarg(4));\n");
+    P("    \n");
+    P("    auto& _r = _get_random_var_by_name(_name, _threadargs_);\n");
+    P("    auto _status = _init_rng(_r, _id1, _id2, _id3);\n");
+    P("    return (_status);\n");
+    P("}\n\n");
+
+    // Hoc wrapper for sample_rng function
+    P("static double _hoc_sample_rng(void* _vptr) {\n");
+    common_code_for_hoc_function();
+    P("    if (!ifarg(1)) {\n");
+    P("        hoc_execerror(\"sample_rng() requires 1 argument!\", NULL);\n");
+    P("    }");
+    P("    const char *_name = gargstr(1);\n");
+    P("    auto& _r123state = _get_random_var_by_name(_name, _threadargs_);\n");
+    // Iterate through random variables and call corresponding sample function
+    for (const auto& var: get_random_variables()) {
+        Fprintf(fcout, "    if(strcmp(_name, \"%s\") == 0) {\n", var.name.c_str());
+        const auto& method = var.get_random123_function_for_distribution();
+        // create arguments with comma separation
+        auto args_with_comma = [](const std::string& acc, const std::string& str) {
+            return acc.empty() ? str : acc + ", " + str;
+        };
+        std::string result = std::accumulate(var.arguments.begin(),
+                                             var.arguments.end(),
+                                             std::string(),
+                                             args_with_comma);
```

Comment:
I'm a bit confused why this is done here and not in `random_construct.cpp` ? It's just nitpicking, but just wondering.